py/desispec/test/test_proc_night.py

```diff
@@ -287,4 +309,62 @@ def test_proc_night_linking_and_ccdcalib(self):
         for job in ['linkcal', 'psfnight']:
             self.assertTrue(job in proctable['JOBDESC'])
         for job in ['nightlybias', 'ccdcalib', 'nightlyflat']:
-            self.assertTrue(job not in proctable['JOBDESC'])
\ No newline at end of file
+            self.assertTrue(job not in proctable['JOBDESC'])
+
+    @unittest.skipUnless(os.path.isdir(_real_rawnight_dir), f'{_real_rawnight_dir} not available')
+    def test_proc_night_daily(self):
+        """
+        Test proc_night daily mode on nights with partial data
+
+        Requires being at NERSC to inspect input raw data
+        """
+
+        while True:
+            num_newlinks = link_rawdata(self.real_rawnight_dir, self.test_rawnight_dir, numexp=10)
+            desispec.scripts.tile_redshifts.reset_allexp_cache()
+            if num_newlinks == 0:
+                break
+            else:
+                proctable, unproctable = proc_night(self.dailynight, daily=True, still_acquiring=True,
+                                                    z_submit_types=['cumulative',],
+                                                    dry_run_level=1, sub_wait_time=0.0)
+
+
+                etable = load_table(findfile('exposure_table', self.dailynight))
+                keep = etable['LASTSTEP'] != 'ignore'
+                etable = etable[keep]
+
+                ## if 1sec flat has arrived, cals should be submitted, otherwise nothing processed yet
+                has_1secflat = np.any((etable['OBSTYPE']=='flat') & (np.abs(etable['EXPTIME']-1)<0.1))
+                if has_1secflat:
+                    ## if 1sec flat has arrived, cals should be submitted.
+                    ## Note: this could be different if we switch to testing a daily night with
+                    ## and override file, in which case e.g. it could have linkcal instead of nightlyflat
+                    for jobdesc in ('ccdcalib', 'arc', 'psfnight', 'flat', 'nightlyflat'):
```

Comment:
I don't think we want the "pass" here. I will also point out this is only true on nights without an override file in which we aren't linking anything. So if we never change the night we're testing, then the jobdesc's listed here will be present. I'm happy to keep it, with that caveat